src/components/Scorecards/ScorecardsServicePage/ScorecardsServiceRulesFilter.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { FormControl, Select, MenuItem } from '@material-ui/core';
+import React, { useMemo } from 'react';
+import { getRuleTitle } from '../../../utils/ScorecardRules';
+import { StringIndexable } from '../../ReportsPage/HeatmapPage/HeatmapUtils';
+import { Rule, RuleOutcome } from '../../../api/types';
+
+interface ScorecardsServiceRulesFilterProps {
+  selected?: Rule;
+  setSelected: (value?: Rule) => void;
+  rules: RuleOutcome[];
+}
+
+const ScorecardsServiceRulesFilter: React.FC<ScorecardsServiceRulesFilterProps> =
+  ({ selected, setSelected, rules }) => {
+    const rulesMap: StringIndexable<Rule | undefined> = useMemo(() => {
+      return rules.reduce(
+        (acc, curr) => ({
+          ...acc,
+          [getRuleTitle(curr.rule)]: curr.rule,
+        }),
+        { '': undefined },
```

Comment:
lol why not just
```suggestion
        {},
```
?

Reply:
I did it to simplify `onChange` function and to have every possible option in this map. If "all rules" value is selected on dropdown, then `selected` rule should be set to undefined (as we don't want to filter over any specific rule). Let me simplify this and instead of `''`, I'll define it as `'all-rules'`.